File: 프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    answer = 0
    routes.sort(key=lambda x: x[1])
    n = len(routes)
    camera = -float('inf')
    for route in routes:
        car_in, car_out = route
        if car_in > camera:
            answer+=1
            camera = car_out
    return answer

File: 프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(vector<int>& A, vector<int>& B){
    return A[1] < B[1];
}

int solution(vector<vector<int>> routes) {
    int answer = 0;
    sort(routes.begin(), routes.end(), compare);

    int camera = -30001; 
    for (auto& route : routes) {
        if (camera < route[0]) { 
            camera = route[1];   
            answer++;
        }
    }
    return answer;
}